front/src/common/api/generatedEditoastApi.ts

```diff
@@ -3443,6 +3443,22 @@ export type PathfindingResultSuccess = {
   /** Path description as track ranges */
   track_section_ranges: TrackRange[];
 };
+export type OffsetRange = {
+  end: number;
+  start: number;
+};
+export type IncompatibleOffsetRangeWithValue = {
+  range: OffsetRange;
+  value: string;
```

Comment:
![Capture d’écran 2024-08-02 à 17 27 47](https://github.com/user-attachments/assets/7678144d-b212-496a-8689-110a33ea1504)

Request for @bougue-pe : would it be possible to return "non-electrified" instead of an empty string when the range is non electrified ? In the `pathProperties` endpoint, in the `electrification` property, we have this : 

<img width="254" alt="Capture d’écran 2024-08-02 à 17 30 48" src="https://github.com/user-attachments/assets/0b6988a8-0c85-45a7-af6d-f04dc3f57a21">

It might be safer to do that instead of the front assuming that an empty string always means that it's a "non-electrified" behavior. What do you think ?

Reply:
This may be ambiguous no matter what string is chosen as the "no-electrification" representation.
Introducing an option would be the better choice here (so null in that case).
This raises 2 questions I think:
1. Improving this also comes with design questions: how to display clearly the "missing electrification" case?
2. Internally in core this is not a cost we want to pay in this PI (handling options internally might be an important issue with `DistanceRangeMap`). IMO, we should track it as an improvement issue, but I'm not sure that this is going to be judged reasonable in the end.

Reply:
Reply to self 😅 :
1. The display is just "writing it" (from the mockup)
![un-electrified](https://github.com/user-attachments/assets/479a8b9b-4565-4cac-a900-5de3820ee7cc)

And as for the `"non_electrified"` string it is not the same as an option (or enum in the electrification properties)...
We could change the interface, but the internal core model would still be dirty, which is more contained and similar to electrification property.
We could reuse the enum from pathProperties (maybe the same for other constraints if possible).
Leading to a bit heavier but more consistent responses and mutualized code.

Reply:
As discussed with @sim51 : No change in back for the moment, matching in front that empty string to display non-electrified ranges as expected.
Then the issue #8327 previously created would force migration of all this because of the type change.

Reply:
Ok fine by me :)